ext/drwrap/drwrap.h

```diff
@@ -887,6 +889,18 @@ DR_EXPORT
 bool
 drwrap_get_stats(INOUT drwrap_stats_t *stats);
 
+DR_EXPORT
+/**
+ * If the provided app_pc (\p possibly_sentinel) is indeed the address of the
+ * internal replace_retaddr_sentinel(), this routine replaces it with the actual
+ * return address of the inner-most nested wrapped function. Otherwise, it is a no-op.
+ * This allows mitigation of a transparency violation under the #DRWRAP_REPLACE_RETADDR
+ * strategy where the return address on the stack is replaced with the address of the
+ * internal replace_retaddr_sentinel().
+ */
+void
+drwrap_replace_if_retaddr_sentinel(void *drcontext, INOUT app_pc *possibly_sentinel);
```

Comment:
The name is too close to drwrap_replace() and related routines: I think it has to be changed to not start with those 2 words.  `drwrap_update_*` or `drwrap_substitute_*` or sthg?

Reply:
I replaced it with `drwrap_get_retaddr_if_sentinel`: clearly says it only "gets" so no side-effect, also tells what is returned (the retaddr).

Reply:
It's also different than the existing routines you pointed out.

Reply:
> It's also different than the existing routines you pointed out.

The concern is that there are two different drwrap modes: wrap and replace.  The high-level routines for those start with drwrap_wrap* and drwrap_replace*.  Seeing a function starting with drwrap_replace* makes me think it is related to replace mode, which it is not.  `drwrap_get_retaddr_if_sentinel` SGTM.

Reply:
> The concern is that there are two different drwrap modes: wrap and replace.

I  understand.